views.go

```diff
@@ -117,23 +117,91 @@ func ProcessViewResult(result ViewResult, params map[string]interface{},
 	return result, nil
 }
 
-func ReduceViewResult(reduceFunction string, result *ViewResult) error {
+func ReduceViewResult(reduceFunction string, params map[string]interface{}, result *ViewResult) error {
+	reduceFun, compileErr := ReduceFunc(reduceFunction)
+	if compileErr != nil {
+		return compileErr
+	}
+	groupLevel := 0
+	if params["group"] != nil && params["group"].(bool) == true {
+		groupLevel = -1
+	} else if params["group_level"] != nil {
+		groupLevel = int(params["group_level"].(uint64))
+	}
+
+	if groupLevel != 0 {
+		// fmt.Println("groupLevel %v", groupLevel)
+
+		var collator JSONCollator
+
+		k := result.Rows[0].Key
+		if groupLevel != -1 {
+			k = result.Rows[0].Key.([]interface{})[0:groupLevel]
```

Comment:
This code is really hard to understand what's going on.  Maybe if you extract `result.Rows[0].Key.([]interface{})` into a named variable, and then call `[0:groupLevel]` on that.